shoov/modules/custom/shoov_repository/shoov_repository.module

```diff
@@ -167,3 +167,152 @@ function shoov_repository_check_repository_unique($github_id, $node) {
 
   throw new Exception(format_string('GitHub ID @githubid was duplicated.', array('@githubid' => $github_id)));
 }
+
+/**
+ * Check .shoov.yml file exists in repository.
+ *
+ * @param $node
+ *  Repository node
+ *
+ * @return bool
+ *  TRUE if file exists.
+ */
+function shoov_repository_yml_file_exists($node) {
+  if ($node->type != 'repository') {
+    return FALSE;
+  }
+
+  if (variable_get('test_mode', FALSE)){
+    // This is the test mode. Don't check the repo.
+    return TRUE;
+  }
+
+  // Get the master branch tree.
+  $tree = shoov_repository_get_repo_tree($node->title);
```

Comment:
https://github.com/shoov/shoov-nodejs/blob/4ebbfdb30608114ef587017565918ae7b9ba9e92/routes/ci.js#L183

(notice it's branch aware)